File: problems/Task Scheduler/leastInterval.cs
public class Solution {
    public int LeastInterval(char[] tasks, int n) {
        var taskFrequencies = new int[26];

        foreach (var task in tasks) {
            ++taskFrequencies[task - 'A'];
        }

        Array.Sort(taskFrequencies);

        var maxCoolingTimes = taskFrequencies[25] - 1;
        var idleSlotsCount = maxCoolingTimes * n;

        for (var i = 24; 0 <= i; --i) {
            idleSlotsCount -= Math.Min(taskFrequencies[i], maxCoolingTimes);
        }

        return Math.Max(0, idleSlotsCount) + tasks.Length;
    }
}

File: problems/Task Scheduler/leastInterval.java
class Solution {
    public int leastInterval(char[] tasks, int n) {
        int[] taskFrequencies = new int[26];

        for (char task : tasks) {
            ++taskFrequencies[task - 'A'];
        }

        Arrays.sort(taskFrequencies);

        int idleSlotsCounter = n * (taskFrequencies[25] - 1);

        for (int idx = 24; 0 <= idx; --idx) {
            idleSlotsCounter -= Math.min(taskFrequencies[idx], taskFrequencies[25] - 1);
        }

        return tasks.length + Math.max(0, idleSlotsCounter);
    }
}
